massa-execution-worker/src/active_history.rs

```diff
@@ -0,0 +1,105 @@
+use massa_execution_exports::ExecutionOutput;
+use massa_hash::Hash;
+use massa_ledger_exports::{
+    LedgerEntry, LedgerEntryUpdate, SetOrDelete, SetOrKeep, SetUpdateOrDelete,
+};
+use massa_models::{Address, Amount};
+use std::collections::VecDeque;
+
+#[derive(Default)]
+pub(crate) struct ActiveHistory(pub VecDeque<ExecutionOutput>);
+
+pub enum HistorySearchResult<T> {
+    Found(T),
+    NotFound,
+    Deleted,
+}
+
+impl ActiveHistory {
+    /// Lazily query (from end to beginning) the active balance of an address at a given slot.
+    ///
+    /// Returns None if the address balance could not be determined from the active history.
```

Comment:
I think the comment is outdated here. (same for other functions)

Reply:
Fixed.

Reply:
I was speaking about the `Returns None` but the function return an enum not an option.